torch/nn/modules/module.py

```diff
@@ -958,6 +976,25 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
             key = prefix + name
             if key in state_dict:
                 input_param = state_dict[key]
+                # Detect Unitialized parameters or buffers
+                if isinstance(input_param, _UninitializedParameter):
+                    if not isinstance(param, _UninitializedParameter):
+                        raise ValueError('Can\'t load an uninitialized Parameter {} into an initialized one'.format(name))
+                elif isinstance(input_param, _UninitializedBuffer):
+                    if not isinstance(param, _UninitializedParameter):
+                        raise ValueError('Can\'t load an uninitialized Buffer {} into an initialized one'.format(name))
+                if isinstance(param, _UninitializedParameter): 
+                    # The current parameter is not initialized but the being loaded one is
+                    # create a new parameter based on the existing one
```

Comment:
You actually create it based on the one that is given, no the uninitialized param here. Don't you want to change this?
Also to make this more streamlined for this and users implementing Module, do we want a `uninit_param.materialize(size)` method on the UninitializedParameter that will return a Parameter with the given size and the right device/type?

Reply:
Changed :)!